packages/block-library/src/classic/lazy.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+export default class LazyLoad extends Component {
+	static loadedScripts = new Set();
+	static loadedStyles = new Set();
+	static defaultProps = {
+		scripts: [],
+		styles: [],
+		placeholder: null,
+		onLoaded: () => Promise.resolve(),
+		onError: noop,
+	};
+
+	state = {
+		loaded: false,
+	};
+
+	loadScripts = async () => {
+		const scriptsToLoad = this.props.scripts.filter(
+			( script ) => ! this.loadedScripts.has( script )
+		);
+
+		if ( scriptsToLoad.length === 0 ) {
+			return Promise.resolve();
+		}
+
+		await new Promise( ( resolve, reject ) => {
+			const scriptElement = document.createElement( 'script' );
+			// can't use load-scripts.php, so this would need to be replaced
+			scriptElement.src = `/wp-admin/load-scripts.php?load=${ scriptsToLoad.join(
```

Comment:
I know this is a draft, but just a note to remember that `/wp-admin` can be in a subfolder; `example.org/sub/wp-admin`

I don't remember if it's possible to rename it but there's `admin_url` filter.

Reply:
Oh, thanks for pointing that out!